subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -353,16 +349,16 @@ static int hawkbit_device_acid_update(int32_t new_value)
  */
 static void hawkbit_update_sleep(struct hawkbit_ctl_res *hawkbit_res)
 {
-	uint32_t sleep_time;
+	int sleep_time;
 	const char *sleep = hawkbit_res->config.polling.sleep;
 
 	if (strlen(sleep) != HAWKBIT_SLEEP_LENGTH) {
 		LOG_ERR("Invalid poll sleep: %s", sleep);
 	} else {
 		sleep_time = hawkbit_time2sec(sleep);
-		if (sleep_time > 0 && poll_sleep != (MSEC_PER_SEC * sleep_time)) {
+		if (sleep_time > 0 && poll_sleep != sleep_time) {
```

Comment:
same here re: error handling, might be worth handling better the case where `hawkbit_time2sec` returns -1?